godot-core/src/obj/script.rs

```diff
@@ -523,6 +546,41 @@ mod bounded_ptr_list {
             unsafe { Box::from_raw(slice) }
         }
     }
+
+    /// Convert a list into a pointer + length pair. Should be used together with [`list_from_sys`](list_from_sys).
+    ///
+    /// If `list_from_sys` is not called on this list then that will cause a memory leak.
+    #[cfg(since_api = "4.3")]
+    pub fn list_into_sys<T>(list: Vec<T>) -> (*const T, u32) {
+        let len: u32 = list
+            .len()
+            .try_into()
+            .expect("list must have length that fits in u32");
+        let ptr = Box::leak(list.into_boxed_slice()).as_mut_ptr();
+
+        (ptr.cast_const(), len)
+    }
+
+    /// Get a list back from a previous call to [`list_into_sys`](list_into_sys).
+    ///
+    /// # Safety
+    /// - `ptr` must have been returned from a call to `list_into_sys`.
+    /// - `ptr` must not have been used in a call to this function before.
+    /// - `ptr` must not have been mutated since the call to `list_into_sys`.
+    /// - `ptr` must not be accessed after calling this function.
+    #[cfg(since_api = "4.3")]
+    #[deny(unsafe_op_in_unsafe_fn)]
+    pub unsafe fn list_from_sys<T>(ptr: *const T, len: u32) -> Box<[T]> {
+        let ptr: *mut T = ptr.cast_mut();
+        let len: usize = sys::conv::u32_to_usize(len);
+
+        // SAFETY: `ptr` was created in `list_into_sys` from a slice of length `len`.
+        // And has not been mutated since.
+        let slice = unsafe { std::slice::from_raw_parts_mut(ptr, len) };
```

Comment:
Should maybe also mention why the `*const` -> `*mut` conversion is sound.

(It was mutable in the first place, but GDExtension API requires const ptr).